### DISCUSSION TITLE
Play 2.8.20

### DISCUSSION THREAD

**Discussion by mkurz** [playframework/playframework#11886] [Announcements]
The Play Team is happy to announce the release of Play 2.8.20 :partying_face:

## :grey_question: Nice, but what about Play 2.9?

It's around the corner, basically some docs, migration guide and release notes missing, but no more pull requests. A first release candidate should follow within days :crossed_fingers: Sorry, it took a bit longer and was postponed a couple of times. With the first Play 2.9 RC we will also give an update on how we continue with Akka / Akka HTTP after September 2023.

## :green_book: So what is new in 2.8.20?

This is mostly a bugfix release with some enhancements added. The most notable are:

- #11881 [2.8.x] scala-xml 1.3.1 by @mkurz
  - Ships with a fix to avoid a potential `StackOverflowError` in Play body parsers. Researched by @beny23, thanks!
- #11754 [2.8.x] Add to RouteInvokers test helper (backport #11375) by @dwhjames
- #11876 [2.8.x] Support trailing newline in SSR (backport #11848) by @ryoppy
- #11657 [2.8.x] Support MariaDB jdbc con. string for Evolutions tables (backport #11655) by @mkurz
- #11732 [2.8.x] Add ConfigLoader implicits for java.time.Duration (backport #11715) by @tmccombs
- #11611 [2.8.x] Filters should be restrictive when their route modifier black-/whitelist both are empty (backport #11610) by @mkurz
- #11874 [2.8.x] Update akka to 2.6.21 (backport #11849) by @xuwei-k

Besides that, following pull requests got merged for this release:

- #11878 [2.8.x] Dependency upgrades for 2.8.20 release by @mkurz
- #11877 [2.8.x] Make it easy to identify the cookie in which the error occurred (backport #11850) by @tsuyoshizawa
- #11818 [2.8.x] Fixed dead link (backport #11817) by @gianick
- #11807 [2.8.x] Fix documentation of scheduleOnce (backport #11806) by @johnsgp
- #11802 [2.8.x] Remove asset finder from helloWorld by @mkurz
- #11795 [2.8.x] Sync GHA workflows with main branch by @mkurz
- #11779 [2.8.x] fix reverse routes escape with fixed param (backport #11774) by @xuwei-k
- #11746 [2.8.x] Fix MysqlJdbcUrl regex matching (backport #11745) by @mkurz
- #11713 [2.8.x] Fix transforming of PathBindable and also the anyValPathBindable macro (backport #11712) by @mkurz
- #11668 [2.8.x] Added .bsp to the typical .gitignore file by @gianick
- #11642 [2.8.x] sbt-java-formatter 0.8.0 by @ihostage
- #11639 [2.8.x] Scalafmt3 (backport #11512) by @adrianlyjak
- #11638 [2.8.x] Upgrade Scalafmt to latest release of `2.7.x` serias by @ihostage
- #11636 [2.8.x] Fix length validation in Multipart.randomBoundary (backport #11635) by @jdayton3
- #11634 [2.8.x] Upgrade Scalafmt to latest release of `2.6.x` serias by @ihostage
- #11632 [2.8.x] 🔄 synced file(s) with playframework/.github by @ihostage
- #11630 [2.8.x] Upgrade Scalafmt to latest release of `2.5.x` serias (backport #11628) by @ihostage
- #11627 [2.8.x] implements #11467. Adds command to fix up formats (backport #11513) by @adrianlyjak
- #11626 [2.8.x] Render sha and dirty part in version only when on CI (backport #11625) by @mkurz
- #11624 [2.8.x] chore(#11143): set next minor version when being on main branch (backport #11168) by @gaeljw
- #11607 [2.8.x] Interplay 2.1.11 to avoid out of memory with gpg by @mkurz

For more details see the [full list of changes](https://github.com/playframework/playframework/compare/2.8.19...2.8.20/) and the [2.8.20 milestone](https://github.com/playframework/playframework/issues?page=1&q=milestone%3A%222.8.20%22).

## :heart: Thanks to our premium sponsors!

<div align="center">
  <a href="https://informaticon.com/"><img src="https://www.playframework.com/assets/images/home/sponsors/49a3d5258c8e8b1daf5481eb9e00b898-informaticon-logo-black.png" width="250"></a>
  <a href="https://cedarlakeventures.com/"><img src="https://www.playframework.com/assets/images/home/sponsors/bec2b526c9ce52c051f9089a10044867-cedar-lake-ventures.png" width="250"></a>
  <a href="https://nulab.com/"><img src="https://www.playframework.com/assets/images/home/sponsors/6152e584aa8625eedca1c4accf8f8b63-nulab_logo_color.png" width="250"></a>
  <a href="https://pronto.net/"><img src="https://www.playframework.com/assets/images/home/sponsors/c77b1d664f10a1c9cb19b97c6d8bd204-pronto-software.png" width="250"></a>
  <a href="https://civiform.us/"><img src="https://www.playframework.com/assets/images/home/sponsors/cb047b3782866c962c4d6a35b056b809-civiform.png" width="250"></a>
  <a href="https://theguardian.com/"><img src="https://www.playframework.com/assets/images/home/sponsors/b15eb0f249dbc45089872e268d8ea5ad-the_guardian.png" width="250"></a>
</div>

If you find this OSS project useful for work, please consider asking your company to support it by <a href="https://www.playframework.com/sponsors">becoming a sponsor</a>.
You can also individually sponsor the project by <a href="https://www.playframework.com/sponsors">becoming a backer</a>.

<div align="center">
  <a href="https://opencollective.com/playframework" target="_blank">
    <img src="https://opencollective.com/playframework/donate/button@2x.png?color=blue" width="250" />
  </a>
</div>

## :bow: Thanks to our contributors

Finally, thanks to the community for their help with detailed bug reports, discussions about new features and pull request reviews. This project is only possible due to the help we had from amazing contributors.
Special thanks to all code contributors who helped with this particular release (they are listed below)!

<hr /><em>This discussion was created from the release <a href='https://github.com/playframework/playframework/releases/tag/2.8.20'>Play 2.8.20</a>.</em>

**Comment by brunoballekens** [comment]
Small question (sorry to put it here - not sure where else to put it): did something went wrong when publishing the sbt-plugin ?
We're using scala-steward for our dependency upgrades, but did not yet receive any PR's.
When i looked in it's logs it does not find a new version of the sbt plugin, although the artifact is on maven central (https://repo1.maven.org/maven2/com/typesafe/play/sbt-plugin_2.12_1.0/2.8.20/).
However I believe that scala-steward uses `maven-metadata.xml` to discover the versions and i see that it has not been updated (see https://repo1.maven.org/maven2/com/typesafe/play/sbt-plugin_2.12_1.0/), where for e.g. the main artifact (https://repo1.maven.org/maven2/com/typesafe/play/play_2.13/) it has been updated...

**Reply by mkurz** [reply]
AFAIK scala-steward relies on coursier to determine the versions. Now looking at this [code here](https://github.com/coursier/coursier/blob/v2.1.5/modules/core/shared/src/main/scala/coursier/maven/MavenRepositoryInternal.scala#L123-L124), if a `maven-metadata.xml` is found it will use it, if it is not found, it will list the contents of the directory to figure out the version. 
The xml in https://repo1.maven.org/maven2/com/typesafe/play/sbt-plugin_2.12_1.0/ however only contains `2.9.0-M6` and `2.9.0-M5` but is missing all 2.8.x releases... so it seems the problem is there.
AFAIK the xml file is generated by maven central server side, so this is not our or sbt's fault (?) sbt does not even generate the xml file (I think this is still the case): https://github.com/sbt/sbt/issues/3879

I would reach out to the sonatype folks and ask what's going on here, according to https://maven.apache.org/repository/central-metadata.html

Related issues:
* https://github.com/typelevel/sbt-typelevel/issues/78
* https://github.com/sbt/sbt/issues/3410 I don't think that's relevent put in sbt 1.9 thinks changed a bit around this topic, but pretty sure is not related to this problem

**Comment by ElVargas95** [comment]
Good morning, we are using Play Framework 2.8.20 for Java, and we are facing an issue with a method that we are consuming through an Ajax client. The problem we are encountering is that when executing the transaction, as it is a heavy transaction, the method reaches a point where it spawns another thread and starts its execution again. This leads to the method's execution being duplicated twice, which is a serious issue in production. What can we do to prevent the server from making these double attempts without notifying it?

**Reply by gaeljw** [reply]
Please open a specific issue or discussion thread with a minimal reproduction case, or at the very least more debugging/troubleshooting details.
